source/lac/trilinos_vector_base.cc

```diff
@@ -372,7 +372,7 @@ namespace TrilinosWrappers
     // is zero on _all_ processors.
     const Epetra_MpiComm *mpi_comm
       = dynamic_cast<const Epetra_MpiComm *>(&vector->Map().Comm());
-    Assert(mpi_comm != 0, ExcInternalError());
+    AssertThrow(mpi_comm != 0, ExcInternalError());
```

Comment:
Other question - do we need a dynamic_cast at that point at all? After all the type should be known at compile time.

Replacing the dynamic_cast with a static_cast would solve both problems, (a) no runtime error condition whether the cast succeeds and (b) no `AssertThrow` (which is the wrong macro for an internal error).